vumi_message_store/api/tests/test_message_export_worker.py

```diff
@@ -0,0 +1,476 @@
+        yield worker.startService()
+        port = yield worker.services[0]._waitingForPort
+        addr = port.getHost()
+
+        self.url = 'http://%s:%s' % (addr.host, addr.port)
+
+        self.query_store = worker.store
+        # TODO: There should be a better way to get the other message store
+        # interfaces.
+        riak = self.query_store.manager
+        redis = self.query_store.redis
+        self.operational_store = OperationalMessageStore(riak, redis)
+        self.batch_manager = MessageStoreBatchManager(riak, redis)
+
+        self.addCleanup(self.stop_server, port)
+
+    def stop_server(self, port):
+        d = port.stopListening()
+        d.addCallback(lambda _: port.loseConnection())
+        return d
+
+    def make_batch(self, tag):
+        return self.batch_manager.batch_start([tag])
+
+    def make_outbound(self, batch_id, content, timestamp=None):
+        if timestamp is None:
+            timestamp = datetime.utcnow()
+        msg = self.msg_helper.make_outbound(content, timestamp=timestamp)
+        d = self.operational_store.add_outbound_message(msg,
+                                                        batch_ids=[batch_id])
+        d.addCallback(lambda _: msg)
+        return d
+
+    def make_inbound(self, batch_id, content, timestamp=None):
+        if timestamp is None:
+            timestamp = datetime.utcnow()
+        msg = self.msg_helper.make_inbound(content, timestamp=timestamp)
+        d = self.operational_store.add_inbound_message(msg,
+                                                       batch_ids=[batch_id])
+        d.addCallback(lambda _: msg)
+        return d
+
+    def make_request(self, method, batch_id, leaf, **params):
+        url = '%s/%s/%s/%s' % (self.url, 'resource_path', batch_id, leaf)
+        if params:
+            url = '%s?%s' % (url, urlencode(params))
+        return http_request_full(method=method, url=url)
+
+    def get_batch_resource(self, batch_id):
+        return self.store_resource.getChild(batch_id, None)
+
+    def assert_csv_rows(self, rows, expected):
+        self.assertEqual(sorted(rows), sorted([
+            row_template % {
+                'id': msg['message_id'],
+                'ts': msg['timestamp'].isoformat(),
+            } for row_template, msg in expected
+        ]))
+
+    @inlineCallbacks
+    def test_get_invalid_path(self):
+        """
+        An HTTP request to the server to an unrouted path should return a 404
+        response code.
+        """
+        yield self.start_server()
+        url = "%s/%s" % (self.url, 'bad_path',)
+        resp = yield http_request_full(method='GET', url=url)
+
+        self.assertEqual(resp.code, http.NOT_FOUND)
+
+    @inlineCallbacks
+    def test_get_invalid_batch(self):
+        """
+        An HTTP request to the server for a non-existant batch should return a
+        404 response code.
+        """
+        # TODO: This test fails. Returns 200 + blank response
```

Comment:
Is there any way to determine if a batch exists before we do a `list_batch...` operation? Because what happens is that we do a `list_batch`, there are no pages to iterate so we just return... and return a 200 response code (as far as I can see).